networkit/test/test_generators.py

```diff
@@ -50,7 +96,86 @@ def testEdgeSwitchingMarkovChainGenerator(self):
                 self.assertEqual(G.degree(u), degseq[u])
 
             break
+
+    """def testEdgeSwitchingMarkovChainGeneratorFit(self):
+        degseq = nk.generators.PowerlawDegreeSequence(2, 100).run().getDegreeSequence(100)
+        g = nk.generators.EdgeSwitchingMarkovChainGenerator(degseq, False, 10)  
+        g.fit(self.G)"""
+
+    def testErdosRenyiGenerator(self):
+        g = nk.generators.ErdosRenyiGenerator(10, 0.15, True)
+        self.assertEqual(g.generate().numberOfNodes(), 10)
+
+    def testErdosRenyiGeneratorFit(self):
+        g = nk.generators.ErdosRenyiGenerator(10, 0.15, True)
+        g.fit(self.G)
+
+    def testHavelHakimiGenerator(self):
+        g = nk.generators.HavelHakimiGenerator([2,2,2,2])
+        self.assertTrue(g.isRealizable()) 
+        self.assertTrue(g.getRealizable())
+
+    def testHavelHakimiGeneratorFit(self):
+        g = nk.generators.HavelHakimiGenerator([2,2,2,2])  
+        g.fit(self.G)  
+
+    def testHyperbolicGenerator(self):
+        g = nk.generators.HyperbolicGenerator(100)
+        g.setLeafCapacity(5)
+        g.setBalance(0.5)
+        g.setTheoreticalSplit(False)
+        self.assertEqual(g.generate().numberOfNodes(), 100)
+
+    def testLFRGenerator(self):
+        g = nk.generators.LFRGenerator(5)
+        g.setDegreeSequence([0,0,0,0,0])
+        g.setMu(1.5)
+        g.setCommunitySizeSequence([1,1,1,1,1])
+        self.assertEqual(g.generate().numberOfNodes(), 5)
+        g.run()
+        self.assertEqual(g.getGraph().numberOfEdges(), 0)
+        self.assertEqual(g.getPartition().numberOfElements(), 5)
+
+    def testLFRGeneratorFit(self):
+        g = nk.generators.LFRGenerator(5)
+        g.setDegreeSequence([0,0,0,0,0])
+        g.setMu(1.5)
+        g.setCommunitySizeSequence([1,1,1,1,1])
+        g.fit(self.G)    
+        
+    def testLFRGeneratorPowerlawDegreeSequence(self):
+        #Example with generatingPowerlaw functions
+        g2 = nk.generators.LFRGenerator(100)
+        g2.generatePowerlawDegreeSequence(10, 20, -2)
+        g2.generatePowerlawCommunitySizeSequence(10, 50, -1)
+        g2.setMu(0.5)
+        self.assertEqual(g2.generate().numberOfNodes(), 100)
+        g2.run()
+        self.assertEqual(g2.getGraph().numberOfNodes(), 100)
+        self.assertEqual(g2.getPartition().numberOfElements(), 100)
+
+    def testLFRGeneratorPowerlawDegreeSequenceFit(self):
+        g2 = nk.generators.LFRGenerator(10)
+        g2.generatePowerlawDegreeSequence(5, 6, -2)
+        g2.generatePowerlawCommunitySizeSequence(5, 6, -1)
+        g2.setMu(0.5)
+        g2.fit(self.G, vanilla=True, plfit=False)
```

Comment:
Same as above (`fit` function). Also, better split the fit-test for `LFRGenerator` if different parameters use different lines of code.